src/server/req_manager.c

```diff
@@ -3364,6 +3448,60 @@ setup_ping(int delay)
 
 /**
  * @brief
+ *		mgr_sched_create - process request to create a sched
+ *
+ *		Creates sched
+ *
+ *  @param[in]	preq	- Pointer to a batch request structure
+ */
+static void
+mgr_sched_create(preq)
+struct batch_request *preq;
+{
+	int bad;
+	char *badattr;
+	svrattrl *plist;
+	pbs_sched *psched;
+	int rc;
+
+	if (strlen(preq->rq_ind.rq_manager.rq_objname) > PBS_MAXSCHEDNAME) {
+		req_reject(PBSE_SCHED_NAME_BIG, 0, preq);
+		return;
+	}
+	if (find_scheduler(preq->rq_ind.rq_manager.rq_objname)) {
+		req_reject(PBSE_SCHEDEXIST, 0, preq);
+		return;
+	}
+	if (validate_sched_attr_vals(preq, NULL))
```

Comment:
Doing it in action function is over-head. Action function are on attribute structure, if the validation fails then we need to reject the batch_request. To do so it is better to do validation on svrattrlist